### PR TITLE
Improve docstring of Annoation

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

As part of #10148: Docstring update for `Annotation`.

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -1986,107 +1986,115 @@ class Annotation(Text, _AnnotationBase):
...
 
         Parameters
         ----------
-
         s : str
             The text of the annotation.
 
-        xy : iterable
-            Length 2 sequence specifying the *(x,y)* point to annotate.
+        xy : 2-tuple of float
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/text.py]: I don't think it needs to be a 2-tuple, and neither a float, e.g. the following are all valid

```
a = mtext.Annotation("String", xy=[.5,.2], xytext=np.array([.65,.32]))
b = mtext.Annotation("String", xy=range(2), xytext={3,5})
```

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -1986,107 +1986,115 @@ class Annotation(Text, _AnnotationBase):
...
-            If a `~matplotlib.transforms.Transform` object is passed
-            in use that to transform ``xy`` to screen coordinates
+                def transform(renderer) -> Bbox
+                def transform(renderer) -> Transform
 
-            If a callable it must take a
-            `~matplotlib.backend_bases.RendererBase` object as input
-            and return a `~matplotlib.transforms.Transform` or
-            `~matplotlib.transforms.Bbox` object
+              where *render* is a `.RendererBase` subclass.
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/text.py]: *render**er***

File: lib/matplotlib/text.py [hunk 3]
```diff
@@ -2124,25 +2132,32 @@ def __init__(self, s, xy,
...
 
-            The default is `None`, which behave as `True` only if
-            *xycoords* is "data".
+        **kwargs
+            Additional kwargs are passed to `~matplotlib.text.Text`.
 
         Returns
         -------
-        Annotation
+        `.Annotation`
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/text.py]: This renders as 
image

File: lib/matplotlib/text.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/text.py]: I'm aware of this.

#### float

We've decided to
> Use float for a type that can be any number.

https://matplotlib.org/devel/documenting_mpl.html#parameter-type-descriptions

https://github.com/matplotlib/matplotlib/issues/10225#issuecomment-360975571

#### tuple

While iterable is technically correct, it lacks all the relevant information (required fixed size of 2 and type). Technically, you would have to write `iterable of float of length 2` or `length-2 iterable of float`, which is IMO is not quite readable and scary to less experienced programmers. Lot's of other docstrings use tuple as well even though you can pass other iterables. I'm willing to sacrifice precision of the definition for simplicity here.

We do not have a consistent and good way of describing this (should be added to the [documentation recommendations](https://matplotlib.org/devel/documenting_mpl.html#parameter-type-descriptions) once we've reached a decision). Some other options used in the code base (modulo float <-> scalar) are:

- `tuple of 2 floats`
- `(float, float)`
- `tuple of (float, float)`

Thinking again, maybe `(float, float)` is simple and clear enough. It concisely defines contained type and size, and does not explicitly mention the word tuple.

**Comment by timhoffm** [comment]
I've added docstrings for `anncoords` and `xyann`. It's unfortunate that they have a different name than the corresponding constructor parameters. But that would be something for another PR.

**Comment by anntzer** [comment]
See also https://github.com/matplotlib/matplotlib/pull/12144 re: mismatched names in constructor.

**Comment by ImportanceOfBeingErnest** [comment]
> I've added docstrings for anncoords and xyann.

I don't see any change to the previous version. 
image

**Comment by ImportanceOfBeingErnest** [comment]
This was tagged with v3.0.x. So when merging, it tried to autobackport to that branch. I resolved the conflict (which was a single empty line (??)), and created #12214, but does this now still need to go to v3.0.0-doc? Should it have been tagged with v3.0.0-doc instead?

**Pull request merged**